runelite-client/src/main/java/net/runelite/client/plugins/tithefarm/TitheFarmPlantOverlay.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.tithefarm;
+
+import net.runelite.api.Client;
+import net.runelite.api.widgets.Widget;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.awt.*;
+import java.awt.geom.Arc2D;
+import java.util.HashSet;
+import java.util.Set;
+
+public class TitheFarmPlantOverlay extends Overlay
+{
+	private static final int TIMER_SIZE = 25;
+	private static final int TIMER_BORDER_WIDTH = 1;
+
+	private final Client client;
+	private final TitheFarmPlugin plugin;
+	private final TitheFarmPluginConfig config;
+
+	private Color colorUnwatered;
+	private Color colorWatered;
+	private Color colorGrown;
+
+	@Inject
+	TitheFarmPlantOverlay(@Nullable Client client, TitheFarmPlugin plugin, TitheFarmPluginConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.UNDER_WIDGETS);
+		this.plugin = plugin;
+		this.config = config;
+		this.client = client;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		if (config.enabled())
+		{
+			checkPlants();
+			drawPlants(graphics);
+		}
+		return null;
+	}
+
+	public void updateColors()
+	{
+		colorGrown = config.getColorGrown();
+		colorWatered = config.getColorWatered();
+		colorUnwatered = config.getColorUnwatered();
+	}
+
+	private void checkPlants()
+	{
+		Set<TitheFarmPlant> plantsToRemove = new HashSet<>();
+		for (TitheFarmPlant plant : plugin.getPlants())
+		{
+			if (plant.getPlantTimeRelative() == 1)
+			{
+				plantsToRemove.add(plant);
```

Comment:
Why are you not calling plugin.removePlant(plant) directly here?

Reply:
It creates warnings about changing the set while iterating through it. If it's okay to get those warnings I can change it back.

Reply:
plugin.getPlants().removeIf(plant -> plant.getPlantTimeRelative() == 1)